testsuite/features/support/network_utils.rb

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2024 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+require 'net/scp'
+require 'net/ssh'
+require 'stringio'
+
+Net::SSH::Transport::Algorithms::ALGORITHMS.each_value { |algs| algs.reject! { |a| a.match(/^ecd(sa|h)-sha2/) } }
+Net::SSH::KnownHosts::SUPPORTED_TYPE.reject! { |t| t.match(/^ecd(sa|h)-sha2/) }
```

Comment:
I think you can use `=~` instead of `match()` again since we reverted to Ruby 3.1 again.

Reply:
If it works, let's keep, otherwise it will spoy again once we upgrade Ruby.

Reply:
Sure, fine with me.